src/unix/haiku/bsd.rs

```diff
@@ -0,0 +1,151 @@
+use crate::prelude::*;
+
+// This file contains the BSD APIs available in Haiku. It corresponds to the
+// header files in `headers/compatibility/bsd`.
+//
+// Note that Haiku's BSD compatibility is a combination of system APIs and
+// utility libraries. There should only be system APIs in `libc`. When you are
+// trying to determine whether something should be included in this file, the
+// best indicator is whether it also exists in the BSD-specific definitions in
+// this libc crate.
```

Comment:
Would you mind turning this into `//!` module doc comments since it's moving anyway?